homeassistant/components/withings/config_flow.py

```diff
@@ -46,33 +48,60 @@ async def async_oauth_create_entry(self, data: dict) -> dict:
 
     async def async_step_profile(self, data: dict) -> dict:
         """Prompt the user to select a user profile."""
-        profile = data.get(const.PROFILE)
+        errors = {}
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        context_profile = self.context.get(const.PROFILE)
+        profile = data.get(const.PROFILE) or context_profile
 
         if profile:
-            new_data = {**self._current_data, **{const.PROFILE: profile}}
-            self._current_data = None
-            return await self.async_step_finish(new_data)
+            existing_entry = next(
+                iter(
+                    config_entry
+                    for config_entry in self.hass.config_entries.async_entries(
+                        const.DOMAIN
+                    )
+                    if slugify(config_entry.data.get(const.PROFILE)) == slugify(profile)
+                ),
+                None,
+            )
+
+            if context_profile or not existing_entry:
+                new_data = {**self._current_data, **{const.PROFILE: profile}}
+                self._current_data = None
+                return await self.async_step_finish(new_data)
+
+            errors["base"] = "profile_exists"
 
-        profiles = self.hass.data[const.DOMAIN][const.CONFIG][const.CONF_PROFILES]
         return self.async_show_form(
             step_id="profile",
-            data_schema=vol.Schema({vol.Required(const.PROFILE): vol.In(profiles)}),
+            data_schema=vol.Schema({vol.Required(const.PROFILE): str}),
+            errors=errors,
         )
 
-    async def async_step_reauth(self, data: dict) -> dict:
+    async def async_step_reauth(self, data: Optional[Dict]) -> dict:
         """Prompt user to re-authenticate."""
         if data is not None:
             return await self.async_step_user()
 
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        placeholders = {"profile": self.context["profile"]}
+
+        self.context.update({"title_placeholders": placeholders})
+
         return self.async_show_form(
             step_id="reauth",
             # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
-            description_placeholders={"profile": self.context["profile"]},
+            description_placeholders=placeholders,
         )
 
     async def async_step_finish(self, data: dict) -> dict:
         """Finish the flow."""
         self._current_data = None
 
-        await self.async_set_unique_id(data["token"]["userid"], raise_on_progress=False)
+        config_entry = await self.async_set_unique_id(
+            str(data["token"]["userid"]), raise_on_progress=False
+        )
+        if config_entry:
+            await self.hass.config_entries.async_remove(config_entry.entry_id)
```

Comment:
Why do we remove the existing entry instead of aborting before creating a new entry?

Is this after a reauth step? Then we should update the existing entry instead. We can do that by using `self._abort_if_unique_id_configured` and pass the `updates` parameter.

Reply:
Will do. Question. How do I set the message for the abort message? After running reauth, a empty dialog saying "Abort" is displayed. Adding a message to strings.json and translations/en.json abort > already_configured doesn't help.

Reply:
That should work. Maybe refresh the browser?